#VRML_SIM R2022a utf8
WorldInfo {
  basicTimeStep 5
}
Viewpoint {
  orientation -0.013748300468770605 -0.013571353062530283 -0.9998133838923505 1.5839366176378034
  position 0.003500617169555853 3.3046511016652604 0.714758927708357
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorTileSize 0.25 0.25
  wallHeight 0.05
}
Robot {
  translation 0 0 0.05
  children [
    DEF Base Solid {
      translation 0 0 0.05
      children [
        Shape {
          appearance Asphalt {
          }
          geometry Box {
            size 2 2 0.1
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 1.49
            axis 0 1 0
          }
          endPoint Solid {
            translation -4.163336342344337e-17 0 0.441
            rotation 0 1 0 0
            children [
              InertialUnit {
                rotation 0 0 1 1.5707996938995747
                name "imu"
              }
              HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0 0 0.2448
                  minStop -1.0472
                  maxStop 1.0472
                }
                device [
                  RotationalMotor {
                    name "balance_motor"
                    controlPID 0.0001 0 0
                    maxTorque 100
                  }
                  PositionSensor {
                    name "pendulum_position"
                  }
                ]
                endPoint Solid {
                  translation 0 0 0.65
                  rotation 0 1 0 0
                  children [
                    Shape {
                      appearance GlossyPaint {
                        baseColor 0.4 1 1
                      }
                      geometry Sphere {
                        radius 0.1
                      }
                    }
                  ]
                  name "rider"
                  boundingObject Sphere {
                    radius 0.1
                  }
                  physics Physics {
                    mass 7
                  }
                }
              }
              Shape {
                appearance GalvanizedMetal {
                }
                geometry Sphere {
                  radius 0.1
                }
              }
            ]
            name "frame"
            boundingObject Sphere {
              radius 0.1
            }
            physics Physics {
              mass 39.2
            }
          }
        }
      ]
      name "base"
      boundingObject Shape {
        geometry Box {
          size 2 2 0.1
        }
      }
      physics Physics {
        mass 10
      }
    }
  ]
  boundingObject Box {
    size 2 2 0.01
  }
  physics Physics {
  }
  controller "Pendulum_Controller"
}
